napari/_qt/qt_viewer.py

```diff
@@ -1151,3 +1217,43 @@ def _create_remote_manager(
     qt_poll.events.poll.connect(monitor.on_poll)
 
     return manager
+
+
+def _get_reader_choice_for_file(
+    readerDialog: Any, readers: Dict[str, str], error_message: str
+) -> Tuple(str, bool):
+    """Gets choice of reader from user for the given filename.
+
+    If an existing association is set for this extension, tries
+    to open with that reader, and pops choice dialog if that fails.
+
+    If no existing association is set, pop the choice dialog and
+    get a chosen reader from the users.
+
+    Parameters
+    ----------
+    readerDialog : QtReaderDialog or MockQtReaderDialog
+        reader dialog to use for choices from the user
+    readers: str
+        Dictionary of display-name:plugin-name of potential readers for file
+
+    Returns
+    -------
+    display_name: str
+        Display name of the chosen plugin
+    persist_choice: bool
+        Whether to persist the chosen plugin choice or not
+    """
+    display_name = ''
+    persist_choice = False
+
+    if len(readers) == 1 and not error_message:
```

Comment:
Since this is the only use of `error_message` in here, should probably replace it with a bool parameter like `is_ok` or `encountered_error`. 

You've done a nice job of documenting the function but the `error_message` parameter isn't described.  A bool parameter might be simpler to doc.